test/hack/bin/fetch-tools.sh

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# let's have reproducable tests - pin to specific versions
+ARCHITECT_VERSION="6.5.0"
+PROMETHEUS_VERSION="2.36.2"
+HELM_VERSION="3.9.0"
+YQ_VERSION="4.26.1"
+
+GIT_WORKDIR=$(git rev-parse --show-toplevel)
+
+OS_BASE="$(uname -s)"
+case "${OS_BASE}" in
+Linux*)
+    export PROMETHEUS_SOURCE="https://github.com/prometheus/prometheus/releases/download/v${PROMETHEUS_VERSION}/prometheus-${PROMETHEUS_VERSION}.linux-amd64.tar.gz"
+    export HELM_SOURCE="https://get.helm.sh/helm-v${HELM_VERSION}-linux-amd64.tar.gz"
+    export ARCHITECT_SOURCE="https://github.com/giantswarm/architect/releases/download/v${ARCHITECT_VERSION}/architect-v${ARCHITECT_VERSION}-linux-amd64.tar.gz"
+    export YQ_SOURCE="https://github.com/mikefarah/yq/releases/download/v${YQ_VERSION}/yq_linux_amd64.tar.gz"
+    export YQ_BIN_FILE="yq_linux_amd64"
+    ;;
+
+Darwin*)
+    export PROMETHEUS_SOURCE="https://github.com/prometheus/prometheus/releases/download/v${PROMETHEUS_VERSION}/prometheus-${PROMETHEUS_VERSION}.darwin-amd64.tar.gz"
+    export HELM_SOURCE="https://get.helm.sh/helm-v${HELM_VERSION}-darwin-amd64.tar.gz"
+    export ARCHITECT_SOURCE="https://github.com/giantswarm/architect/releases/download/v${ARCHITECT_VERSION}/architect-v${ARCHITECT_VERSION}-darwin-amd64.tar.gz"
+    export YQ_SOURCE="https://github.com/mikefarah/yq/releases/download/v${YQ_VERSION}/yq_darwin_arm64.tar.gz"
+    export YQ_BIN_FILE="yq_darwin_arm64"
+    ;;
+
+*)
+    echo "${OS_BASE} not supported - if needed, please implement by your own."
+    ;;
+
+esac
```

Comment:
How about we simply into this

```suggestion
Linux*)
    os="linux"
    ;;

Darwin*)
    os="darwin"
    ;;

*)
    echo "${OS_BASE} not supported - if needed, please implement by your own."
    exit 1
    ;;

esac

export PROMETHEUS_SOURCE="https://github.com/prometheus/prometheus/releases/download/v${PROMETHEUS_VERSION}/prometheus-${PROMETHEUS_VERSION}.${OS}-amd64.tar.gz"
    export HELM_SOURCE="https://get.helm.sh/helm-v${HELM_VERSION}-${OS}-amd64.tar.gz"
    export ARCHITECT_SOURCE="https://github.com/giantswarm/architect/releases/download/v${ARCHITECT_VERSION}/architect-v${ARCHITECT_VERSION}-${OS}-amd64.tar.gz"
    export YQ_SOURCE="https://github.com/mikefarah/yq/releases/download/v${YQ_VERSION}/yq_${OS}_amd64.tar.gz"
    export YQ_BIN_FILE="yq_${OS}_amd64"
```

Reply:
Would not work as-is because some packages are arm64 for Darwin.
May be a problem though; we should maybe rely on the actual machine's architecture.

Reply:
Somehow solved, or I do not care enough.